modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/BaseOrderItemResourceImpl.java

```diff
@@ -752,9 +753,33 @@ public void create(
 			"createStrategy", "INSERT");
 
 		if ("UPSERT".equalsIgnoreCase(createStrategy)) {
-			orderItemUnsafeConsumer =
-				orderItem -> putOrderItemByExternalReferenceCode(
-					orderItem.getExternalReferenceCode(), orderItem);
+			String updateStrategy = (String)parameters.getOrDefault(
+				"updateStrategy", "UPDATE");
+
+			if ("UPDATE".equals(updateStrategy)) {
+				orderItemUnsafeConsumer =
+					orderItem -> putOrderItemByExternalReferenceCode(
+						orderItem.getExternalReferenceCode(), orderItem);
+			}
+
+			if ("PARTIAL_UPDATE".equals(updateStrategy)) {
+				orderItemUnsafeConsumer = orderItem -> {
+					try {
+						OrderItem getOrderItem =
+							getOrderItemByExternalReferenceCode(
+								orderItem.getExternalReferenceCode());
+
+						patchOrderItem(
+							getOrderItem.getId() != null ?
+								getOrderItem.getId() :
+									_parseLong(
+										(String)parameters.get("orderItemId")),
+							orderItem);
+					}
+					catch (NoSuchModelException noSuchModelException) {
+					}
```

Comment:
I think in this case, an exception should be thrown. Otherwise I think the import task could finished successfully without being imported

Reply:
Fixed https://github.com/liferay-headless/liferay-portal/pull/1246/commits/e149468ca203e6534c4f950797edc1a8bca96fe6#diff-65c62040644c8c1625c30add3374509d558af7334a3c313f488e6ba4d3efa0f0R678